### DISCUSSION TITLE
Folding boxes/foldables

### DISCUSSION THREAD

**Discussion by daaronr** [quarto-dev/quarto-cli#467] [Q&A] [answered]
It's very helpful to be able to hide and fold content, not just code.  This allows you to hide details only some readers care about, and to do so in a longer-form and richer way than hover footnotes.

We implement this in our own style of Bookdown, as demonstrated [HERE](https://rethinkpriorities.github.io/bookdown-template/foldable-sections.html), and I've seen and done this in other Rmd styles.

If quarto could add this as a feature it would make it much easier for me and the teams I'm working with to switch over!

**Comment by dragonstyle** [comment]
This is a good suggestion, thank you. We'll add it to the todo list for the future. We have our hands pretty full right now, so not sure exactly when we'll get to it...

**Comment by jjallaire** [comment]
@daaronr We will definitely implement something like this -- seems quite broadly useful. As @dragonstyle this not be for a few weeks though. In the meantime I think it would be pretty easy to implement this as a Lua filter that targets `.foldable` divs or as just some Javascript/CSS that's added in an `include` (not sure how you did this for bookdown but my guess is that the solution would look nearly identical for Quarto)

**Comment by jjallaire** [comment, marked as answer]
Another note: you could implement a variation of this this today using collapsible callouts, see https://quarto.org/docs/authoring/callouts.html#collapse

**Reply by daaronr** [reply]
Collapsible callouts seems like the thing I'm looking for. Shall I close this issue?

**Reply by jjallaire** [reply]
Since this isn't actually a formal issue (it's an "Idea" in Discussions) we can't actually close it, but yes let's consider this one resolved!

**Reply by madebyafox** [reply]
I would put a big upvote on having foldable sections (perhaps defined at some # depth of the TOC) beyond that provided in collapsible callouts! :)

**Reply by dragonstyle** [reply]
This would be a great case for an extension (we just opened up docs on extending Quarto with filters and shortcodes). Would be cool to implement a filter that collapses sections below a certain level- you could like just use bootstrap collapse attributes to inject toggles and collapse the sections.

Read more about extensions here:

https://quarto.org/docs/extensions/

**Comment by mghassany** [comment]
I was doing this using `details` and `summary` in bookdown with somme css. Thanks for the collapsable callouts ! They do the job perfectly.

**Comment by mcanouil** [comment]
Additionally:

- https://quarto.org/docs/authoring/callouts.html#collapse
- https://github.com/shafayetShafee/collapse-callout
